common/src/chain/pow.rs

```diff
@@ -0,0 +1,112 @@
+use crate::chain::config::ChainType;
+use crate::chain::ChainConfig;
+use crate::Uint256;
+use std::time::Duration;
+
+/// Chain Parameters for Proof of Work, as found in
+/// https://github.com/bitcoin/bitcoin/blob/eca694a4e78d54ce4e29b388b3e81b06e55c2293/src/chainparams.cpp
+pub struct PoWChainConfig {
+    no_retargeting: bool,
+    allow_min_difficulty_blocks: bool,
+    limit: Uint256,
+}
+
+impl ChainConfig {
+    pub fn get_pow_config(&self) -> PoWChainConfig {
```

Comment:
So every time we call this function, we'll instantiate a new one with all the relevant values? Kind of suboptimal. Can you think of a better way to do this?

Reply:
would you prefer converting the whole `ChainConfig` into `PoWChainConfig` ?

Reply:
The problem here is the constant reservation of memory on every call to `get_pow_config`. I don't have a specific solution in mind. These values should be reserved once, only when the ChainConfig created. Then, they should be requested on demand.

Reply:
I am not convinced optimising this is worth the effort. If this is done only once per block, it's rather trivial operation compared to the rest of the processing of the block. I'd just leave as it is and only revisit if it actually shows up during profiling (which I quite doubt it will). Also compilers tend to be really good at optimizing this stuff, all the temporaries will likely be gone when you compile this.

Reply:
Well, it's just bad practice. Sure, the compiler may be nice enough to fix this, but why play that game if a better design can come with this? For proof of stake, this is going to be much bigger and more complex. It's a win if Carla can come up with a better way to do this for all consensus algorithms. Besides, this is included in the first line of defense where we verify the work of a block (assuming we're proof of work chain, but again, we're seeking things to be done nominally right), hence, any spam attempt will face this first. Sure, we shouldn't spend a week planning this. But since we're in the design stage, let's aim for this to be done right.

Reply:
updated; converted method into `const fn`